### PR TITLE
chore: update detectron2 example to use v0.6 and reenable nightly test [MLG-301]

### PR DESCRIPTION
**Pull request by drh-determined-ai**
## Description

The `nightly` test `test_capability.py::test_detectron2_coco_pytorch_const` was marked to be skipped during the attempt to upgrade default Determined images to use PyTorch 1.12 (see [this PR](https://github.com/determined-ai/determined/pull/5985)).

We have created new images with the latest version of Detectron2, v0.6, updated the experiment configurations accordingly, and made minimal compatibility changes to the experiment code.

## Test Plan

`test-e2e-gpu-nightly` will be approved and is expected to be successful.

## Commentary (optional)

The previous custom image used in the Detectron2 example (found [here](https://hub.docker.com/r/katport/detectron2)) was built with Detectron2 v0.1.2. 

The new image is built with v0.6, the latest version, which was released in October 2021. The image and example were manually QA'd by the author on a GCP instance with two (2) K80 GPUs. 

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by drh-determined-ai** [comment]
Successful `test-nightly` run, including `test_capability.py::test_detectron2_coco_pytorch_const` [here](https://app.circleci.com/pipelines/github/determined-ai/determined/34132/workflows/69aabf3c-1987-48c3-91b9-db92249945ad/jobs/1205265).

**Comment by drh-determined-ai** [comment]
Note that `test-stress` was [a bit flaky](https://app.circleci.com/pipelines/github/determined-ai/determined/34132/workflows/f6bc8d3a-1f56-4ab7-a559-6cefc9fb93d6/jobs/1205199): it succeeded on [the second run](https://app.circleci.com/pipelines/github/determined-ai/determined/34132/workflows/bfbcb400-bf2e-497e-baa4-014c9d6b2103/jobs/1205794).

File: examples/computer_vision/detectron2_coco_pytorch/model_def.py [hunk 1]
```diff
@@ -73,8 +74,8 @@ def build_validation_data_loader(self):
         return data_loader
 
     def train_batch(self, batch: TorchData, epoch_idx: int, batch_idx: int):
-
-        loss_dict = self.model(batch)
+        with EventStorage() as storage:
```
Comment (mpkouznetsov) [inline examples/computer_vision/detectron2_coco_pytorch/model_def.py:77]: if you are not actually using `storage` you can omit `as storage` and just say `with EventStorage():`

**Review by mpkouznetsov — APPROVED** [review]
Looks good.

**Pull request merged**